db/versions/0091-downgrade.sql

```diff
@@ -0,0 +1,6 @@
+begin
+  drop table queue_pending_tasks;
+  drop table queue_claimed_tasks;
+  drop table queue_resolved_tasks;
+  drop table queue_task_deadlines;
```

Comment:
This is a bit of a cheat -- if, for some reason, the migration succeeds but there's some issue that leads you to downgrade an installation, that installation's queues will be out of sync with the task tables. As written, the queues will contain "stale" data from the moment the upgrade transaction committed, while the task tables will contain up-to-date information about tasks. So for example you might have pending tasks for which there's no entry in the pending queue.

I know it seems like a nightmare of SQL that is unlikely to ever be needed, but I think it's a good idea to write a full downgrade process that reconstructs `azure_queue_messages` from these four tables, using an exclusive table lock just like `0091-migration.sql`.

Reply:
Great point! 

I think it's easy to put deadline/claim/resolve back, but for pending you need to create queue name which is bit more tricky, but I agree, data is lost on downgrade.
Will add 👍